chat-android/src/main/java/com/ably/chat/Connection.kt

```diff
@@ -112,6 +113,13 @@ public interface Connection {
     }
 }
 
+/**
+ * @return [ConnectionStatusChange] events as a [Flow]
+ */
+public fun Connection.statusAsFlow(): Flow<ConnectionStatusChange> = transformCallbackAsFlow {
```

Comment:
Since, this returns a cold flow and not shared, it can only be used by one collector.
I think this should be made explicit in the doc or method name itself ( that this is not a shared flow ).

Reply:
Public API can get confusing where developers will treat it as a shared flow and apply collectors on a single flow instance. In such a case, we might like to rename those methods 
```
// rename `statusAsFlow` to
statusAsColdFlow
```
or update method doc
```
@return [ConnectionStatusChange] events as a cold [Flow]. This flow can't be shared.
```
Same goes for usage at other places

Reply:
No, there is no such restrictions, it can be collected as many times as user wants. `SharedFlow` is `Flow` itself and it's clear from the return type that we return `Flow` :)

Reply:
"Cold" and "Hot" terminology useful for developers that creates flow, not for end users of the Flow. If we had shared flow internally and expose it nothing would've changed for end users (although it would've been "Hot" flow)